### DISCUSSION TITLE
Should a global configuration supported?

### DISCUSSION THREAD

**Discussion by babyfish-ct** [babyfish-ct/jimmer#419] [Ideas]
![8837c5d31e4a0a37c4c8503003ce8e50](https://github.com/babyfish-ct/jimmer/assets/15168900/55b48cac-9a7c-4a02-ad18-04792b2ff409)

Now, if entity of either side of `@JoinTable` supports logical deletion, the `@JoinTable` itself must also support logical deletion by specifying `@JoinTable.logicalDeletedFilter`.

If you don't want specify `logicalDeletedFilter`, `deletedWhenEndpointIsLogicalDeleted` which means rows of middle table will be physically deleted when either side if logically deleted must be specified, otherwise, exception will be raised.

This solution will let user know what he is doing. However, this solution will affect existing applications, Maybe add a global configuration which can automatically enable `deletedWhenEndpointIsLogicalDeleted` for all join tables should be added.

Let user know what he is doing is important, make there is no affects for existing applications is also important. Which is more important?

**Comment by qinkangdeid** [comment]
是否可以这样
都加配置

1. 想要主表中间表数据一致。中间表也可添加逻辑删除字段  

2. 当主表添加逻辑删除字段 中间表未添加 可选中间表的删除操作 

   2.1  不处理中间表数据 这是现版本的行为 新版保持不变?  也能向下兼容   毕竟数据删了就真没了
 
   2.2 提供关系映射中让用户主动式声明定义物理删除的行为标记  数据删了是开发者的行为

**Reply by babyfish-ct** [reply]
可以
